test/supports/concurrency_config.py

```diff
@@ -0,0 +1,22 @@
+# Copyright Modal Labs 2025
+import modal
+
+app = modal.App("concurrency-config")
+
+CONFIG_VALS = {"OLD": 100, "NEW": 1000}
+
+
+@app.function(allow_concurrent_inputs=CONFIG_VALS["OLD"])
+def has_old_concurrency_config():
+    pass
+
+
+@app.function()
+@modal.concurrent(max_inputs=CONFIG_VALS["NEW"])
+def has_new_concurrency_config():
+    pass
+
+
+@app.function()
+def has_no_concurrency_config():
+    pass
```

Comment:
We should add failing tests for:

```
@concurrent
@method
```

and

```
@method
@concurrent
```

Reply:
And even just this, which might also give unexpected results otherwise:

```py
@app.cls()
class A:
    @concurrent
     def some_method(...):
```

Reply:
Hm yeah we'll need to catch that in `app.cls` I guess? Pretty messy.

Reply:
`Cls.from_local` is probably a good place to put it. We probably have other similar cases that are odd already, like @batched. 

All partial functions on the class must either be FUNCTION, BUILD_FUNCTION, ENTER_X, EXIT-flagged to be valid, I think, anything else would indicate a mistake (the web endpoint class of annotations should add FUNCTION as well)

Reply:
Is FUNCTION supposed to mean "METHOD" or is it valid for decorators that should go on a (regular, lowercase-f) function as well?